pennylane/templates/state_preparations/qrom_state_prep.py

```diff
@@ -0,0 +1,281 @@
+# Copyright 2018-2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""Contains the QROMStatePreparation template."""
+
+import numpy as np
+
+import pennylane as qml
+from pennylane.operation import Operation
+from pennylane.wires import Wires
+
+
+def _float_to_binary(val, num_bits):
+    r"""Converts a value within the range [0, 1) to its binary representation with a specified precision.
+
+    Args:
+        val (float): The value to convert to binary. Must be in the range [0, 1).
+        num_bits (int): the number of bits to use for the binary representation
+
+    Returns:
+        str: The binary representation of the value, with the specified precision.
+
+    **Example**
+
+        >>> _float_to_binary(0.5, 3)
+        '100'
+
+        Expected value as the binary representation of `0.5` is `0.100`.
+    """
+
+    binary_rep = bin(int(2 ** (num_bits + 1) + 2 ** (num_bits) * val))
+    if binary_rep[-num_bits - 1] == "1":
+        return "1" * num_bits
+
+    return binary_rep[-num_bits:]
+
+
+class QROMStatePreparation(Operation):
+    r"""Prepares a quantum state using a Quantum Read-Only Memory (QROM) based approach.
+
+    This operation implements the state preparation method described
+    in `arXiv:0208112 <https://arxiv.org/abs/quant-ph/0208112>`_.
+
+    Args:
+        state_vector (TensorLike): The state vector to prepare.
```

Comment:
Is the type Array or tensorlike? Please make sure you are consistent with other state prep templates that use a state vector.